modules/cognitive-services/src/main.tf

```diff
@@ -0,0 +1,71 @@
+resource "azurerm_cognitive_account" "main" {
+  name                               = "cog-${var.zone}-${var.environment}-${lookup(local.short_locations, var.location)}-${local.identifier}"
+  kind                               = var.kind
+  location                           = var.location
+  resource_group_name                = var.resource_group_name
+  sku_name                           = var.sku_name
+  custom_subdomain_name              = var.custom_subdomain_name
+  dynamic_throttling_enabled         = var.dynamic_throttling_enabled
+  fqdns                              = var.fqdns
+  local_auth_enabled                 = var.local_auth_enabled
+  outbound_network_access_restricted = var.outbound_network_access_restricted
+  public_network_access_enabled      = var.public_network_access_enabled
+
+  tags = merge(local.tags, var.tags)
+
+  dynamic "identity" {
+    for_each = var.identity != null ? [var.identity] : []
+    content {
+      type         = identity.value.type
+      identity_ids = identity.value.identity_ids
+    }
+  }
+
+  dynamic "storage" {
+    for_each = var.storage
+    content {
+      storage_account_id = storage.value.storage_account_id
+      identity_client_id = storage.value.identity_client_id
+    }
+  }
+}
+
+resource "azurerm_cognitive_deployment" "main" {
```

Comment:
I don't love the inclusion of another resource in here, it's setting a precedent. Do these deployment resource belong in the workload instead?